portfolio/src/main/webapp/blogScript.js

```diff
@@ -1,27 +1,52 @@
 window.addEventListener('DOMContentLoaded', (event) => {
+    getURL();
+    getPost();
     getComments();
 });
 
-function getComments() {
-    fetch('/data').then(response => response.json()).then((comments) => {  
-      const commentListElement = document.getElementById('comment-list');
-      for (comment of comments) {
-        commentListElement.appendChild(createCommentElement(comment));
+const tagIndx = parent.document.URL.substring(parent.document.URL.indexOf('tag='), parent.document.URL.length);
```

Comment:
Instead of manually parsing a URL, get JS to do it for you! :)

https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

```js
const tag = new URL(document.URL).searchParams.get('tag');
```